extension-manifest-v3/src/utils/engines.js

```diff
@@ -342,6 +387,11 @@ async function update(name) {
 
     if (updated) {
       console.log(`Engine "${name}" updated`);
+
+      // Notify listeners
+      const fns = updateListeners.get(name);
+      fns?.forEach((fn) => fn());
```

Comment:
listener callbacks should be guarded by try/catch - in case any of them throw, the saveToStorage should still be called